File: OpenMP_cpu/simd/4_collapse/collapse.c
#define M 4096
#define N    3
double a[M][N], b[M][N], c[M][N];

void element_mXm(){

// The two loops can be collapsed
 //#pragma omp 
   for (int j=0; j<M; j++) {
      for (int i=0; i<N; i++) {
      c[j][i] = c[j][i] + a[j][i] * b[j][i];
      }
   }
}

File: legacy/OpenMP_cpu/simd/4_collapse/collapse.f90

subroutine element_mxm(a,b,c)
integer,parameter :: N=3, M=4096
double precision,dimension(N,*) :: a,b,c

!! The two loops can be collapsed

 !!$omp 
   do j=1,M
      do i = 1,N
      c(i,j) = c(i,j) + a(i,j) * b(i,j)
   enddo; enddo
end subroutine
